### PR TITLE
Added deployment automation for docs to cloudflare pages

### PR DESCRIPTION
**Pull request by adrienne-deriv**

### CODE REVIEW

File: .github/workflows/build-and-publish-docs.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    group: cloudflare-pages-deploy-docs-${{github.head_ref}}
+    cancel-in-progress: true
+
+jobs:
+    build_to_cloudflare_pages:
+        timeout-minutes: 30
+        runs-on: ubuntu-latest
+        permissions:
+          contents: read
+          deployments: write
```
Comment (adrienne-deriv) [inline .github/workflows/build-and-publish-docs.yml:18]: Need dis to allow Github to show the deployment links in `Release` section

File: .github/workflows/build-and-publish-docs.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            - name: Get cached dependencies
+              id: cache-npm
+              uses: actions/cache/restore@e12d46a63a90f2fae62d114769bbf2a179198b5c
+              with:
+                  path: node_modules
+                  key: npm-docs-${{ hashFiles('./utils-docs/package-lock.json') }}
+
+            - name: Install dependencies
+              if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+              run: cd utils-docs && npm ci
```
Comment (niloofar-deriv) [inline .github/workflows/build-and-publish-docs.yml:39]: For this step can we use shared actions here?
https://github.com/deriv-com/shared-actions/tree/master/.github/actions

File: .github/workflows/build-and-publish-docs.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            - name: Publish to Cloudflare Pages
+                    uses: cloudflare/pages-action@v1
+                    with:
+                      apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+                      accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID}}
+                      projectName: deriv-utils
+                      directory: ./utils-docs/build
+                      gitHubToken: ${{ secrets.GITHUB_TOKEN }}
+                      branch: master
+                      wranglerVersion: '3'
```
Comment (niloofar-deriv) [inline .github/workflows/build-and-publish-docs.yml:53]: Could u pls add an empty line at the end of file for fixing the format of this

File: .github/workflows/build-and-publish-docs.yml (hunk 2 above)
Comment (adrienne-deriv) [reply .github/workflows/build-and-publish-docs.yml:39]: The steps for installing dependencies only install them at the root, we need to install and cache the dependencies at `./utils-docs/`

Comment (niloofar-deriv) [reply .github/workflows/build-and-publish-docs.yml:38]: oh I see so then it's good

**Review by niloofar-deriv — APPROVED** [review]

File: .github/workflows/build-and-publish-docs.yml (hunk 2 above)
Comment (adrienne-deriv) [reply .github/workflows/build-and-publish-docs.yml:38]: lemme update the shared action for CF

**Review by niloofar-deriv — APPROVED** [review]

**Review by jim-deriv — APPROVED** [review]

**Pull request merged**